rfc/0020-group-policies.md

```diff
@@ -0,0 +1,263 @@
+  policies:
+    - name: sigstore_pgp
+      module: ghcr.io/kubewarden/policies/verify-image-signatures:v0.2.8
+      settings:
+        signatures:
+          - image: "*"
+            pubKeys:
+              - "-----BEGIN PUBLIC KEY-----xxxxx-----END PUBLIC KEY-----"
+              - "-----BEGIN PUBLIC KEY-----xxxxx-----END PUBLIC KEY-----"
+    - name: sigstore_gh_action
+      module: ghcr.io/kubewarden/policies/verify-image-signatures:v0.2.8
+      settings:
+        signatures:
+          - image: "*"
+            githubActions:
+            owner: "kubewarden"
+    - name: reject_latest_tag
+      module: ghcr.io/kubewarden/policies/trusted-repos-policy:v0.1.12
+      settings:
+        tags:
+          reject:
+            - latest
+  expression: "sigstore_pgp() || (sigstore_gh_action() && reject_latest_tag())"
+  message: "The group policy is rejected."
+```
+
+### Expression evaluation
+
+Unfortunately, no production-ready CEL library is available for Rust.
+To emulate the behavior and the syntax of CEL, we will use [Rhai](https://rhai.rs/), a small, fast, easy-to-use scripting language and evaluation engine that integrates tightly with Rust.
+
+Rhai has the following features that make it a good candidate for this task:
+
+- function calls have the same syntax as CEL
+- it is easy to embed in a Rust application
+- it provides a way to create a [minimal engine](https://rhai.rs/book/engine/raw.html) with a bare minimum set of functionalities
+- it does not add a considerable overhead (see [Preliminary benchmark](#preliminary-benchmark))
+- the language supports [syntax customization](https://rhai.rs/book/engine/custom-syntax.html) if needed
+
+Considering the following expression from the example above:
+
+```rust
+sigstore_pgp() || (sigstore_gh_action() && reject_latest_tag())
+```
+
+Each policy in the group will be represented as a function call in the expression with the same name as the policy defined in the group.
+
+The expression will be evaluated as follows:
+
+- the `sigstore_pgp` function will be called
+- if the result is `true`, the expression will be evaluated as `true`
+- if the result is `false`, the `sigstore_gh_action` function will be called
+- if the result is `true`, the `latest_tag` function will be called
+- if the result is `true`, the expression will be evaluated as `true`
+
+Note that the `sigstore_gh_action` function will be called only if the `sigstore_pgp` function returns `false`.
+Similarly, the `reject_latest_tag` function will be called only if the `sigstore_gh_action` function returns `true`.
+
+This avoids evaluating all the policies in the group when the result is already known.
+
+### Settings validation
+
+When the policy server starts, it will validate the settings of the group policies as well as the single policies.
+However, the group policies will have an additional validation step to ensure that the expression is valid and
+that evaluates to a boolean value.
```

Comment:
I think we should also look for a misconfiguration like mixing policies that look into totally different resources. For example, there's no sense in doing `policy_that_eval_ingress() && policy_that_eval_pods()`.

In this case I think we should print a big fat warning